### PR TITLE
[s] My Little Station: April Fools is Magic

### PR DESCRIPTION
**Pull request by Core0verload**
This PR adds ponies as April Fools only (triggered by AF event) roundstart playable race. Yes, it uses holiday datums. Also there is a prob(15) that "human" race will switch to "pony" at pref save load. Because I just want to see the server burn.
pony
Unlike Goof's sanics, it actually works without breaking anything.

Ponies have a mutant color, colored hairstyles, eyes and tails. Styles are chosen randomly, color matches the hair color. There is a small chance to spawn with wings, alicorn, or both. Feel free to add those choices to player panel, I was simply too lazy to do that for an one-day event. Ponies hide clothing icons on them to avoid icons breaking, but don't hide belt or inhands. Ponies are banned from heads.

Ponies have a "pure friendship" reagent in them instead of blood. When injected in human it switches it's intent to help, displays random friendly messages and sometimes makes the human hug nearby people. 
Attention: friendship can be addictive and corruptive!

The reason I did this is, first, some guy sent me link to the [ponystation git](https://github.com/S-Spy/PonyBuild), and second, ponies tend to create extreme levels of controversy around them. They are like skub, but everyone knows what they are without google.

This PR requires no config changes and will only enable itself on April Fools. All changed prefs will be reverted at the end of event.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
how come yours doesn't have like 200 commits but mine does

**Comment by Xhuis** [comment]
Jesus Christ it's horrible

**Comment by Iamgoofball** [comment]
Go fuck yourself for both the comments in the OP and the feature

**Comment by Incoming5643** [comment]
@PKPenguin321 Base the pull off the april fools day branch instead of the master

on this...

God

File: code/modules/holiday/aprilfools.dm [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+///////////
+// PROCS //
+///////////
+
+/datum/species/pony/on_species_gain(mob/living/carbon/C)
+	C.reagents.add_reagent("friendship", 100)
+	for(var/addiction in C.reagents.addiction_list)
+		var/datum/reagent/R = addiction
+		if(R.id == "friendship")
+			C.reagents.addiction_list.Remove(addiction)
```
Comment (Iamgoofball) [inline code/modules/holiday/aprilfools.dm]: uh

**Comment by Bawhoppen** [comment]
:point_up:x:100:

**Comment by Incoming5643** [comment]
No seriously the level of familiarity I'm seeing with this stuff is legitimately worrying me. I think on some design level this was made as something **that wasn't shitposting**

**Comment by Core0verload** [comment]
I stole the sprites from there: https://github.com/S-Spy/PonyBuild. 
LICENSING MEMES: yep, they have a compatible license.

**Comment by Zilenan91** [comment]
Oh...

**_Oh JESUS_**

File: code/modules/holiday/aprilfools.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/holiday/aprilfools.dm]: Yes, friendship is one of the most horrifying drugs known to humanity.

**Comment by Zilenan91** [comment]
Wait there was a fucking pony server...

**Comment by Steelpoint** [comment]
My only regret is we don't have dismemberment alongside these things.

**Comment by Core0verload** [comment]
@Steelpoint, ponystation actually HAS pony dismemberment sprites. So if by the next April Fools we'll have dismemberment in, maybe i'll try this again.

Also, we still have flamethrowers and gibbers.

**Comment by Zilenan91** [comment]
So ponystation is an actual thing? Do they have a site? I need to see this out of a sense or morbid curiosity.

**Comment by PKPenguin321** [comment]
github needs a "physically ill" reaction tag

**Comment by Core0verload** [comment]
AFAIK no, they are not public and have no public server. I don't even know if they are still up, because the repo looks abandoned.

**Comment by Bawhoppen** [comment]
have you looked at most lesser server's repos? they're always dead

**Comment by Zilenan91** [comment]
Yeah I drudged it up. Site was Ponystation13.com

Thing died about ~3 years ago.

**Comment by Steelpoint** [comment]
Surprisingly nice poster, just need to add a reflective colour to the helmet visor and cut out the rear part.

File: code/modules/client/preferences_savefile.dm [hunk 2]
```diff
@@ -295,7 +295,11 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	//Species
 	var/species_id
 	S["species"]			>> species_id
-	if(config.mutant_races && species_id && (species_id in roundstart_species))
+	if((species_id == "human" || !species_id ) && (SSevent.holidays && SSevent.holidays[APRIL_FOOLS]) && prob(15))
```
Comment (Supermichael777) [inline code/modules/client/preferences_savefile.dm]: i think you accidentalied round-start race restrictions

File: code/modules/holiday/aprilfools.dm [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+/datum/species/pony/handle_speech(message, mob/living/carbon/human/H)
+	return message
+
+//return a list of spans or an empty list
+/datum/species/pony/get_spans()
+	return list()
+
+
+// FRIENDSHIP REAGENT
+// Sto- Ported from Goonstation.
```
Comment (Supermichael777) [inline code/modules/holiday/aprilfools.dm]: GOON CODE IS NON COMPATIBLE WITH OUR LICENCE

File: code/modules/client/preferences_savefile.dm (hunk 2 above)
Comment (Core0verload) [inline code/modules/client/preferences_savefile.dm]: No, it just got moved. See line 302.

File: code/modules/holiday/aprilfools.dm [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+		for (var/mob/living/hugTarget in orange(1,M))
+			if (hugTarget == M)
+				continue
+			if (!hugTarget.stat)
+				M.visible_message("<span class='rose'>[M] [prob(5) ? "awkwardly side-" : ""]hugs [hugTarget]!</span>")
+				break
+
+	if(current_cycle >= 75 && prob(33) && ishuman(M))
+		M << "<span class='userdanger rose'>The friendship overcomes you!</span>"
+		M.set_species(/datum/species/pony)
```
Comment (Supermichael777) [inline code/modules/holiday/aprilfools.dm]: absolutely monsterous

File: code/modules/holiday/aprilfools.dm (hunk 3 above)
Comment (Core0verload) [inline code/modules/holiday/aprilfools.dm]: **LICENSING MEMES**
also nobody gives a fuck anymore, we already have stocks stolen and shipping datums are coming in hot.

Comment (Bawhoppen) [inline code/modules/holiday/aprilfools.dm]: im going to steal content from other games then

Comment (Core0verload) [inline code/modules/holiday/aprilfools.dm]: Start with reverting slimes to metroids.

File: code/modules/holiday/aprilfools.dm (hunk 4 above)
Comment (Core0verload) [inline code/modules/holiday/aprilfools.dm]: Yes, pure friendship is one of the most horrifying drugs known to humanity.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 5]
```diff
@@ -62,6 +62,9 @@ Please contact me on #coderbus IRC. ~Carnie x
 /mob/living/carbon/human/update_damage_overlays()
 	remove_overlay(DAMAGE_LAYER)
 
+	if(dna && istype(dna.species, /datum/species/pony))
```
Comment (KazeEspada) [inline code/modules/mob/living/carbon/human/update_icons.dm]: snowflake

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 6]
```diff
@@ -99,6 +102,9 @@ Please contact me on #coderbus IRC. ~Carnie x
 /mob/living/carbon/human/proc/update_augments()
 	remove_overlay(AUGMENTS_LAYER)
 
+	if(dna && istype(dna.species, /datum/species/pony))
```
Comment (KazeEspada) [inline code/modules/mob/living/carbon/human/update_icons.dm]: snowflake

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 7]
```diff
@@ -162,6 +168,9 @@ Please contact me on #coderbus IRC. ~Carnie x
 				w_uniform.screen_loc = ui_iclothing //...draw the item in the inventory screen
 			client.screen += w_uniform				//Either way, add the item to the HUD
 
+		if(dna && istype(dna.species, /datum/species/pony))
```
Comment (KazeEspada) [inline code/modules/mob/living/carbon/human/update_icons.dm]: you get the idea... theres a better way to do this.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 8]
```diff
@@ -192,11 +201,15 @@ Please contact me on #coderbus IRC. ~Carnie x
 
 /mob/living/carbon/human/update_inv_wear_id()
 	remove_overlay(ID_LAYER)
+
 	if(wear_id)
 		if(client && hud_used && hud_used.hud_shown)
 			wear_id.screen_loc = ui_id
 			client.screen += wear_id
 
+		if(dna && istype(dna.species, /datum/species/pony))
```
Comment (KazeEspada) [inline code/modules/mob/living/carbon/human/update_icons.dm]: it involves using some sort of species flag to tell the icon updating to just fuck off.

File: code/modules/holiday/aprilfools.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/modules/holiday/aprilfools.dm]: shipping datums are original tho

although yeah stocks already got in (with permission from marq, who coded it)

**Comment by Iamgoofball** [comment]
# hasbro gonna sue our asses

File: code/modules/mob/living/carbon/human/update_icons.dm (hunk 8 above)
Comment (Core0verload) [inline code/modules/mob/living/carbon/human/update_icons.dm]: It's a race that would be used only once, for one day. I could make it less snowflake, and I know how to do this, but I'm not going to bother actually doing this. Unless /tg/station would want to keep ponies permanently, which clearly wouldn't happen.

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/update_icons.dm]: i sure hope not but then again we also said lizards weren't ever going to not be admin only and that genetic catbeast ears wouldn't ever be enabled by default

Comment (Steelpoint) [inline code/modules/mob/living/carbon/human/update_icons.dm]: There is no fucking way Ponies are going to be used outside of a meme event or April Fools (at best).

There's a difference between a Lizardperson, who at least has a lot of precedent (Books and Tabletop games) for existing but also somewhat fits in with the universe, and a bunch of cartoon ponies.

Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/update_icons.dm]: s l i p p e r y s l o p e

it stopped being one when it all came true

**Comment by Durkel** [comment]
This triggers the autism.

**Comment by ExplosiveCrate** [comment]
Can we keep the friendship reagent? Because that shit sounds hilarious.

**Comment by Core0verload** [comment]
If we would keep it, how it could be obtained?

**Comment by TrustyGun** [comment]
it should be extracted from humans with the Smile gene (the one that makes you happy)

**Comment by Cheridan** [comment]
oops my finger slipped

**Pull request merged**